tactics/class_tactics.mli

```diff
@@ -57,6 +57,22 @@ val autoapply : constr -> Hints.hint_db_name -> unit Proofview.tactic
 
 val resolve_tc : constr -> unit Proofview.tactic
 
+type solver = (Environ.env ->
+    Evd.evar_map ->
+    Constr.metavariable option ->
+    bool ->
+    best_effort:bool ->
+    (Evd.evar_map -> Evar.t -> bool) ->
+    (bool * Evd.evar_map) option)
+
```

Comment:
```suggestion
type solver = (Environ.env ->
    Evd.evar_map ->
    int option -> (* depth *)
    bool -> (* unique *)
    best_effort:bool ->
    (Evd.evar_map -> Evar.t -> bool) -> (* filter *)
    (bool * Evd.evar_map) option)

```
you could also use labels. You should also document the return type, what is the bool BTW?